### ISSUE TITLE
During training, what's the usage of validation set?

### ISSUE THREAD

**Issue by WhCanI** [ultralytics/ultralytics#18034]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello,

I’ve recently split my dataset into training and validation sets and I am currently training a model using Ultralytics. I would like to clarify the role of the validation set during the training process.

Is the validation set solely used for evaluating the model’s performance after each epoch (i.e., to assess metrics like loss, accuracy, etc.), or is it also involved in model weight updates and backpropagation? In other words, does the validation set play any role in the optimization of the model parameters during training?

Thank you for your help!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @WhCanI, thank you for reaching out and for your interest in Ultralytics 🚀! We recommend checking out the [Docs](https://docs.ultralytics.com) for detailed guidance on training and validation processes, including an overview of [dataset preparation](https://docs.ultralytics.com/datasets/) and [model training](https://docs.ultralytics.com/modes/train/).

Regarding your ❓ Question, an Ultralytics engineer will assist you soon to clarify the role of the validation set during training. In the meantime:

- If you are running experiments locally, ensure you're using the latest version of the `ultralytics` package. You can upgrade it (along with its required dependencies) in a [**Python>=3.8**](https://www.python.org/) environment with:

```bash
pip install -U ultralytics
```

- You might also find value in reviewing our [Training Tips](https://docs.ultralytics.com/guides/model-training-tips/) guide for best practices when splitting datasets and configuring training processes.

🧪 If you are conducting a specific experiment and encounter unexpected behavior, feel free to share additional details such as your dataset splits, training parameters, and logs to help us provide more targeted support.

## Join the Community

Need more ideas or want to discuss your question further? Collaborate with the Ultralytics community where it suits you best:
- Live chat on [Discord](https://discord.com/invite/ultralytics) 🎧
- Thoughtful discussions on [Discourse](https://community.ultralytics.com)
- Engage in threads on our [Subreddit](https://reddit.com/r/Ultralytics)

For validated environments to run your training sessions, you can also explore:
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are passing, ensuring the integrity of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) in verified environments.

Let us know if there’s anything else you’d like to clarify while we work on your specific query! 😊

**Comment by Y-T-G** [comment]
It's only for validating. Model's weights are not updated based on it. It can affect the best.pt indirectly as the performance of that model is determined based on it.

**Comment by WhCanI** [comment]
But I have encountered an issue during model training with Ultralytics. After training a model on a specific dataset, I adjusted the ground truth in the validation set to improve its performance significantly. However, when I retrained the model using the same training set and hyperparameters, but with the adjusted ground truth in the validation set, the new model performed much worse on the validation set compared to the initial training result.

**Comment by Y-T-G** [comment]
The performance of `last.pt` or `best.pt`?

**Comment by Y-T-G** [comment]
Did you delete labels.cache after modification?

**Comment by WhCanI** [comment]
The performance of last.pt, and I delete the labels.cache after modification. But I can't get to know why the performence of the model deteriorated.

**Comment by WhCanI** [comment]
sorry, it is the best.pt

**Comment by Y-T-G** [comment]
Is the performance of last.pt the same as before?

**Comment by WhCanI** [comment]
not exactly the same, but very close

**Comment by Y-T-G** [comment]
The best.pt was probably selected on a different epoch based on the new validation set and so it has different performance.

**Comment by WhCanI** [comment]
But the best.pt performed worse than before on the modified validation set

**Comment by WhCanI** [comment]
the modified validation set is adjusted according to the validation result of the previous model, so I doubt that the validation set may be used for weight calculation or sth else

**Comment by Y-T-G** [comment]
What about the performance on the previous validation set with the new best.pt?

**Comment by WhCanI** [comment]
Slightly worse than the previous model

**Comment by WhCanI** [comment]
The difference of the validation set perhaps influenced the training stage

**Comment by Y-T-G** [comment]
It shouldn't influence it. Do the loss graphs look different?
